### PR TITLE
Infrastructure diagrams/documentation

### PR DESCRIPTION
**Pull request by Stephen-ONeil**

### CODE REVIEW

File: docs/infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+- legacy "data bundles" come from two scripts in `client/build_code`, `copy_static_assets.js` and `write_footnote_bundles.js`
+- `index-{variant}.html` files are not refreshed as part of the standard deploy, due to their legacy management outside of our CDN (in the production case). As a result, the resources directly requested from the html entry file are hardcoded too
+  - `app-{variant}.min.js` must never be cached. It is the entry to the deploy-specific bundle . In practice it _is_ cached at the cloudflare level for performance; cached versions are flushed via the cloudflare API at deploy-time
+  - other hardcoded index html resources may have relatively short client TTLs for performance, but are also flushed from cloudflare during deploys. Tradeoff that changes to them will not immediately propogate to all users
+- `app-{variant}.min.js` needs to be as minimal as possible, it should only start the initial loading spinner and then kick off additional loading of the run time bundles. This gets the spinner going asap and makes up for the limited caching for this file
+
+### GraphQL API (+ batching and caching details)
```
Comment (sranc046) [inline docs/infrastructure.md:208]: Flagging this because there's an error with the diagram rendering

File: docs/infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  - I've kicked the ball down the road on picking one over the other; until more of the app migrates to GraphQL we won't have a full picture on what is called for
+    - if most routes send lots of small queries than batching might be best; provided we can set up the apollo cache to persist to browser storage or something between sessions
+    - if there aren't too many individual queries shooting off at once, HTTP caching is more reliable than alternatives. We could even consider altering the infrastrucutre to gain a static address for our prod API and fronting it through a service like cloudflare, in which case the GET responses (with approrpaite cache busting controls) could get very nice distributed caching
+- excluded from the diagram, but GraphQL resolvers that make mongo queries likely make use of data loaders internally as an additional caching layer. As the cloud functions should be considered ephemeral, this is more about queries that may revisit the same
+
+### Form Backend API
```
Comment (sranc046) [inline docs/infrastructure.md:279]: Diagram rendering error

Comment (sranc046) [inline docs/infrastructure.md:279]: Diagram rendering error

File: docs/infrastructure.md (hunk 1 above)
Comment (sranc046) [inline docs/infrastructure.md:208]: Flagging this because there's an error with the diagram rendering

**Review by sranc046 — CHANGES_REQUESTED** [review]

File: docs/infrastructure.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+Our CDN domain is resolved by Cloudflare which proxies requests to the corresponding GCloud storage. This gives us:
+
+- distributed caching; potential speed boost and takes load off the cloud storage (performance and cost bonus)
+- fine grained caching rules; see configuration in the Cloudflare dashboard
+- programatic cache resets; see selective Cloudflare cache clearing step in the deploy scripts
+- easy and free HTTPS; Google storage itself only handles HTTP. Should we be concerned that the HTTPS is not end-to-end?
+  - Secrecy is not a concern, all data from the CDN is public
+  - privacy? Guess I don't actually know if Cloudflare includes initial requestor information when populating it's own cache, hm
+  - integrity is still provided for the leg of the trip between cloudflare and the client, which covers the regional/local network boxes. What threat actors will be in a hop between Google and Cloudflare _and_ be willing to risk that access doing anything that threatens integrity? Well, maybe certain network-isolated state actors, hm
```
Comment (sranc046) [inline docs/infrastructure.md:149]: Will need clarifications on my end

Comment (Stephen-ONeil) [reply docs/infrastructure.md:149]: Clarified in meeting.

**Review by sranc046 — APPROVED** [review]

**Pull request merged**